### PR TITLE
add featureFlag  AutoPrunePVC

### PR DESCRIPTION
**Pull request by alinadonisa**

### CODE REVIEW

File: pkg/scale/scale.go [hunk 1]
```diff
@@ -57,8 +57,10 @@ func (s *Scaler) EnsureScale(ctx context.Context, scale uint, gRPCPort int32) er
 	// make use of reclaim policy = delete. A reclaim policy of retain is fine
 	// but will result in wasted money, recycle should be considered unsafe and
 	// is officially deprecated by kubernetes.
-	if err := s.PVCPruner.Prune(ctx); err != nil {
-		return errors.Wrap(err, "initial PVC pruning")
+	if usePVCPruner {
+		if err := s.PVCPruner.Prune(ctx); err != nil {
+			return errors.Wrap(err, "initial PVC pruning")
+		}
```
Comment (chrislovecnm) [inline pkg/scale/scale.go:63]: Put an else here logging that we are not removing the pvc

**Review by chrislovecnm — COMMENTED** [review]
Need a logging line

File: pkg/features/operator_features.go [hunk 2]
```diff
@@ -68,6 +68,11 @@ const (
 	// beta: v1.0
 	// RestartCluster uses crdb binary to generate self signed certifcates
 	ClusterRestart featuregate.Feature = "ClusterRestart"
+	// owner: @alina
+	// alpha: v0.1
+	// beta: v1.0
+	// PVCPruner uses crdb binary to prune PVC on decommission
+	PVCPruner featuregate.Feature = "PVCPruner"
```
Comment (udnay) [inline pkg/features/operator_features.go:75]: Can we change this to `AutoPrunePVC` or something so the name is a bit more descriptive?

File: pkg/scale/scale.go [hunk 3]
```diff
@@ -44,7 +44,7 @@ type Scaler struct {
 // In some cases, it may not be possible to full drain a node. In such cases a
 // ErrDecommissioningStalled will be returned  and the node will be left in a
 // decommissioning  state.
-func (s *Scaler) EnsureScale(ctx context.Context, scale uint, gRPCPort int32) error {
+func (s *Scaler) EnsureScale(ctx context.Context, scale uint, gRPCPort int32, usePVCPruner bool) error {
```
Comment (udnay) [inline pkg/scale/scale.go:47]: /nit `prunePVC`

**Review by udnay — COMMENTED** [review]
Add some unit tests to make sure the guard and/or the feature flag works. 

I'd also explicitly enable this flag in decomission e2e tests

File: pkg/scale/scale.go (hunk 1 above)
Comment (alinadonisa) [reply pkg/scale/scale.go:63]: done

File: pkg/scale/scale.go (hunk 3 above)
Comment (alinadonisa) [reply pkg/scale/scale.go]: done

**Comment by udnay** [comment]
Code changes look good but I'd still like to see a test to make sure this is wired off.

**Comment by alinadonisa** [comment]
> e2e tests

@udnay  I will enable this in e2e tests on this PR but the unit test I will add in another PR. Is it ok for you this?

**Comment by chrislovecnm** [comment]
@alinadonisa let’s get the unit test in here that the feature gate is working. See my last PR with decommission feature gate.

File: pkg/actor/decommission_test.go [hunk 4]
```diff
@@ -30,6 +30,7 @@ import (
 func TestDecommisionFeatureFlag(t *testing.T) {
 	// FeatureGate is currently disabled
 	assert.False(t, utilfeature.DefaultMutableFeatureGate.Enabled(features.Decommission))
+	assert.False(t, utilfeature.DefaultMutableFeatureGate.Enabled(features.AutoPrunePVC))
```
Comment (chrislovecnm) [inline pkg/actor/decommission_test.go:33]: @udnay here is the test that the feature gate is shut off. PTAL so we can merge

Comment (udnay) [reply pkg/actor/decommission_test.go:33]: Sure, lets merge this but I also was looking for a test to make sure that when `AutoPrunePVC` is `false` we never call the pruner. This is fine for now though.

**Review by udnay — APPROVED** [review]
LGTM

**Comment by chrislovecnm** [comment]
> Sure, lets merge this but I also was looking for a test to make sure that when AutoPrunePVC is false we never call the pruner. This is fine for now though.

@alinadonisa please open an issue and work on this unit test

**Pull request merged**